app/src/main/kotlin/com/github/gotify/Repository.kt

```diff
@@ -0,0 +1,320 @@
+package com.github.gotify
+
+import com.github.gotify.api.ClientFactory
+import com.github.gotify.client.api.ApplicationApi
+import com.github.gotify.client.api.MessageApi
+import com.github.gotify.client.model.Application
+import com.github.gotify.client.model.Message
+import com.github.gotify.client.model.Paging
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.SharingStarted
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.combine
+import kotlinx.coroutines.flow.distinctUntilChanged
+import kotlinx.coroutines.flow.map
+import kotlinx.coroutines.flow.stateIn
+import kotlinx.coroutines.launch
+import okhttp3.HttpUrl.Companion.toHttpUrlOrNull
+import retrofit2.awaitResponse
+
+/**
+ * A class that represents the repository.
+ *
+ * @property scope The coroutine scope where the repository will run.
+ * @property baseUrl The base url of the server used to create the full url of the images.
+ * @property applicationApi The application api used to interact with the server.
+ * @property messageApi The message api used to interact with the server.
+ */
+class Repository
+private constructor(
```

Comment:
The previous implementation had info logging for "deleting message with id xy", "Loading more messages for appid" and concrete error logging when the fetching failed e.g "failed requesting messages", "Failed to delete message". These should be readded.

And the `println("JcLog` should be removed.